disassemblers/ofrak_capstone/ofrak_capstone/components.py

```diff
@@ -143,7 +144,7 @@ def __init__(
         super().__init__(resource_factory, data_service, resource_service)
         self._disassembler_service = disassembler_service
 
-    async def analyze(self, resource: Resource, config: None) -> RegisterUsage:
+    async def analyze(self, resource: Resource, config: ComponentConfig = None) -> RegisterUsage:
```

Comment:
This is an incorrect type change made throughout this PR. If you must delete the type variable, it should be the following. But setting the variable type to `ComponentConfig` when the value is `None` is definitely not correct.

```suggestion
    async def analyze(self, resource: Resource, config: Optional[ComponentConfig] = None) -> RegisterUsage:
```

Why was it necessary to get rid of the type variable in the first place? The type variable allows the resource method to have the same type as the component, which is a generic subclass of `AbstractComponent` that uses that type variable.

I'm surprised mypy did not complain about this, but maybe we'll see it when the tests get run in CI?

Reply:
Probably best is just:
```python
    async def analyze(self, resource: Resource, config=None) -> RegisterUsage:
```